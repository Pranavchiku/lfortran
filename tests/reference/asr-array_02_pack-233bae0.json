{
    "basename": "asr-array_02_pack-233bae0",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/array_02_pack.f90",
    "infile_hash": "8d58570ab351e8b1ec04ca39b9a3dd9c099cffdcdf4ea487fe742f35",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-array_02_pack-233bae0.stdout",
    "stdout_hash": "10b6c9fb6fee66fa3a533efeaa080638c1a563a20d027474e9af4d01",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}