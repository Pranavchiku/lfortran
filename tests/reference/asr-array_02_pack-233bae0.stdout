(TranslationUnit
    (SymbolTable
        1
        {
            array_02_pack:
                (Program
                    (SymbolTable
                        2
                        {
                            m:
                                (Variable
                                    2
                                    m
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Integer 4)
                                        [((IntegerConstant 1 (Integer 4))
                                        (IntegerConstant 4 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            p:
                                (Variable
                                    2
                                    p
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Integer 4)
                                        [((IntegerConstant 1 (Integer 4))
                                        (IntegerConstant 4 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    array_02_pack
                    []
                    [(Assignment
                        (Var 2 m)
                        (ArrayConstant
                            (Array
                                (Integer 4)
                                [((IntegerConstant 1 (Integer 4))
                                (IntegerConstant 4 (Integer 4)))]
                                FixedSizeArray
                            )
                            ColMajor
                        )
                        ()
                    )
                    (Assignment
                        (Var 2 p)
                        (IntrinsicArrayFunction
                            Pack
                            [(ArrayPhysicalCast
                                (Var 2 m)
                                FixedSizeArray
                                DescriptorArray
                                (Array
                                    (Integer 4)
                                    [((IntegerConstant 1 (Integer 4))
                                    (IntegerConstant 4 (Integer 4)))]
                                    DescriptorArray
                                )
                                ()
                            )
                            (ArrayPhysicalCast
                                (IntegerCompare
                                    (Var 2 m)
                                    NotEq
                                    (IntegerConstant 0 (Integer 4))
                                    (Array
                                        (Logical 4)
                                        [((IntegerConstant 1 (Integer 4))
                                        (IntegerConstant 4 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ()
                                )
                                FixedSizeArray
                                DescriptorArray
                                (Array
                                    (Logical 4)
                                    [((IntegerConstant 1 (Integer 4))
                                    (IntegerConstant 4 (Integer 4)))]
                                    DescriptorArray
                                )
                                ()
                            )
                            (ArrayPhysicalCast
                                (ArrayConstant
                                    (Array
                                        (Integer 4)
                                        [((IntegerConstant 1 (Integer 4))
                                        (IntegerConstant 4 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ColMajor
                                )
                                FixedSizeArray
                                DescriptorArray
                                (Array
                                    (Integer 4)
                                    [((IntegerConstant 1 (Integer 4))
                                    (IntegerConstant 4 (Integer 4)))]
                                    DescriptorArray
                                )
                                ()
                            )]
                            3
                            (Array
                                (Integer 4)
                                [((IntegerConstant 1 (Integer 4))
                                (IntegerConstant 4 (Integer 4)))]
                                FixedSizeArray
                            )
                            ()
                        )
                        ()
                    )
                    (Print
                        [(Var 2 p)]
                        ()
                        ()
                    )
                    (If
                        (IntrinsicArrayFunction
                            Any
                            [(ArrayPhysicalCast
                                (IntegerCompare
                                    (Var 2 p)
                                    NotEq
                                    (ArrayConstant
                                        (Array
                                            (Integer 4)
                                            [((IntegerConstant 1 (Integer 4))
                                            (IntegerConstant 4 (Integer 4)))]
                                            FixedSizeArray
                                        )
                                        ColMajor
                                    )
                                    (Array
                                        (Logical 4)
                                        [((IntegerConstant 1 (Integer 4))
                                        (IntegerConstant 4 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ()
                                )
                                FixedSizeArray
                                DescriptorArray
                                (Array
                                    (Logical 4)
                                    [((IntegerConstant 1 (Integer 4))
                                    (IntegerConstant 4 (Integer 4)))]
                                    DescriptorArray
                                )
                                ()
                            )]
                            0
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )]
                )
        })
    []
)
